hydra-data/src/main/java/com/addthis/hydra/data/query/QueryElementNode.java

```diff
@@ -188,6 +199,39 @@ public QueryElementNode parse(String tok, MutableInt nextColumn) {
         return this;
     }
 
+    /**
+     * The prefix "+[text]" or "+%[text] is interpreted as a default
+     * value. Extract the default value and remove it from
+     * the input token. Optionally "+[text|hits]" can be used to set
+     * the number of hits on the default node.
+     */
+    private String extractDefaultValue(String tok) {
+        int startDefault = -1;
+        if (tok.startsWith(DEFAULT_NODE)) {
+            startDefault = DEFAULT_NODE.length();
+        } else if (tok.startsWith(DEFAULT_ATTACHMENT)) {
+            startDefault = DEFAULT_ATTACHMENT.length();
+        }
+        int endDefault = tok.indexOf(']');
+        if ((startDefault != -1) && (endDefault != -1)) {
+            int hitsLocation = tok.indexOf('|');
+            if ((hitsLocation >= 0) && (hitsLocation < endDefault)) {
+                defaultHits = Integer.parseInt(tok.substring(hitsLocation + 1, endDefault));
+                defaultValue = tok.substring(startDefault, hitsLocation);
+            } else {
+                defaultValue = tok.substring(startDefault, endDefault);
+            }
+            if (tok.startsWith(DEFAULT_NODE)) {
+                tok = "+" + tok.substring(endDefault + 1);
+            } else if (tok.startsWith(DEFAULT_ATTACHMENT)) {
+                tok = "+%" + tok.substring(endDefault + 1);
+            } else {
+                throw new IllegalStateException("Unexpected state in default value extraction");
```

Comment:
doesn't look like this can be reached as a result of the internal state this method accesses (defaultHits, defaultValue). Therefore, should probably be an illegal argument exception?